### PR TITLE
fix: emit NOASSERTION for copyright text to fix SPDX 2.2 validation failure

### LINKED ISSUE #3346: Invalid SPDX: missing copyright text
**What happened**:

Generated SPDX is invalid, mandatory copyright text is missing

**What you expected to happen**:

SPDX should be valid

**Steps to reproduce the issue**:

```
syft docker:bitnami/mongodb:6.0.6-debian-11-r0 --scope all-layers -o spdx-tag-value@2.2 > mongodb-6.0.6-debian-11-r0.spdx
```

**Anything else we need to know?**:

**Environment**:
- Output of `syft version`:
```
Application: syft
Version:    1.14.1
BuildDate:  2024-10-15T12:50:47Z
GitCommit:  Homebrew
GitDescription: [not provided]
Platform:   darwin/arm64
GoVersion:  go1.23.2
Compiler:   gc
```
- OS (e.g: `cat /etc/os-release` or similar):
macOS 14.7

### PR DESCRIPTION
**Pull request by spiffcs**
# Description

@Fearkin I could not push to your branch since it was on `main` of your fork. I've pushed a commit, checked out a local branch, and kept your commits so you still get the contribution credit.

-  Adds `FileCopyrightText` to SPDX formats

Fixes #3346

- Fixes #3346 related PR #3474 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)

# Checklist:

- [ ] I have added unit tests that cover changed behavior
- [ ] I have tested my code in common scenarios and confirmed there are no regressions
- [ ] I have added comments to my code, particularly in hard-to-understand sections

### CODE REVIEW

File: schema/json/schema-latest.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "$schema": "https://json-schema.org/draft/2020-12/schema",
-  "$id": "anchore.io/schema/syft/json/16.0.18/document",
+  "$id": "anchore.io/schema/syft/json/16.0.19/document",
```
Comment (willmurphyscode) [inline schema/json/schema-latest.json:3]: Can you help me understand why there's a schema change? As I understand it we're not affecting Syft JSON output at all.

**Comment by willmurphyscode** [comment]
I plan to add a test that would have caught this on `spdx-json@2.2` output and then get it merged.

**Comment by willmurphyscode** [comment]
Tests added and snapshots updated. This is ready for re-review.

File: syft/format/internal/testutil/test-fixtures/snapshot/stereoscope-fixture-image-simple.golden
Comment (wagoodman) [inline syft/format/internal/testutil/test-fixtures/snapshot/stereoscope-fixture-image-simple.golden:1]: nit: just a heads up for the future... there are two things that can be updated:

https://github.com/anchore/syft/blob/main/syft/format/spdxjson/encoder_test.go#L20-L21

I think the image did not need to be updated (and rarely needs to be), only the snapshots needed to be updated.

**Review by wagoodman — APPROVED** [review]

**Pull request merged**